src/realm/db.cpp

```diff
@@ -1498,6 +1498,23 @@ bool DB::compact(bool bump_version_number, util::Optional<const char*> output_en
     return true;
 }
 
+void DB::write_copy(StringData path, util::Optional<const char*> output_encryption_key)
+{
+    // group::write() will throw if the file already exists.
+    // To prevent this, we have to remove the file (should it exist)
+    // before calling group::write().
+    File::try_remove(path);
```

Comment:
Is this safe though? Isn't it better to throw and have the user handle it rather than silently delete data they might need?

Reply:
You might be right. During some testing it was annoying that I had to delete the file between runs.

Reply:
Added a "allow overwrite" flag.